src/contracts/yas_pool.cairo

```diff
@@ -84,25 +185,304 @@ mod YASPool {
+                liquidity: cache.liquidity_start
+            };
+
+            //TODO: temporary component syntax
+            let state_tick_bitmap = TickBitmap::unsafe_new_contract_state();
+
+            //TODO: temporary component syntax
+            let mut state_tick = Tick::unsafe_new_contract_state();
+
+            // continue swapping as long as we haven't used the entire input/output and haven't reached the price limit
+            loop {
+                if state.amount_specified_remaining.is_non_zero()
+                    && state.sqrt_price_X96 != sqrt_price_limit_X96 {
+                    break;
+                }
+
+                let step_sqrt_price_start_X96 = state.sqrt_price_X96;
+
+                // TODO: test mut
+                let (mut step_tick_next, step_initialized) =
+                    TickBitmap::TickBitmapImpl::next_initialized_tick_within_one_word(
+                    @state_tick_bitmap, state.tick, self.tick_spacing.read(), zero_for_one
+                );
+
+                // ensure that we do not overshoot the min/max tick, as the tick bitmap is not aware of these bounds
+                if step_tick_next < TickMath::MIN_TICK() {
+                    step_tick_next = TickMath::MIN_TICK();
+                } else if step_tick_next > TickMath::MAX_TICK() {
+                    step_tick_next = TickMath::MAX_TICK();
+                };
+
+                // get the price for the next tick
+                let step_sqrt_price_next_X96 = TickMath::get_sqrt_ratio_at_tick(step_tick_next);
+
+                // compute values to swap to the target tick, price limit, or point where input/output amount is exhausted
+                let (ret_sqrt_price_X96, step_amount_in, step_amount_out, mut step_fee_amount) =
+                    SwapMath::compute_swap_step(
+                    state.sqrt_price_X96,
+                    if (zero_for_one && step_sqrt_price_next_X96 < sqrt_price_limit_X96)
+                        || (!zero_for_one && step_sqrt_price_next_X96 > sqrt_price_limit_X96) {
+                        sqrt_price_limit_X96
+                    } else {
+                        step_sqrt_price_next_X96
+                    },
+                    state.liquidity,
+                    state.amount_specified_remaining,
+                    self.fee.read()
+                );
+
+                state.sqrt_price_X96 = ret_sqrt_price_X96;
+
+                if exact_input {
+                    state
+                        .amount_specified_remaining -=
+                            IntegerTrait::<i256>::new(step_amount_in + step_fee_amount, false);
+                    state.amount_calculated = state.amount_calculated
+                        - IntegerTrait::<i256>::new(step_amount_out, false);
+                } else {
+                    state
+                        .amount_specified_remaining +=
+                            IntegerTrait::<i256>::new(step_amount_out, false);
```

Comment:
Would it make sense to have implemented `Into` from u256 to i256?

Reply:
✅